libtransmission/net.c

```diff
@@ -658,7 +658,7 @@ static int tr_globalAddress(int af, void* addr, int* addr_len)
         return -1;
     }
 
-    if (global_unicast_address((struct sockaddr*)&ss) == 0)
+    if (!global_unicast_address((struct sockaddr*)&ss))
```

Comment:
This still just checks for a return of zero-or-nonzero, just as before.

The PR description mentions checking for -1 as well as for 0. Did you mean something like:

```suggestion
    if (global_unicast_address((struct sockaddr*)&ss) <= 0)
```

Reply:
That is what I mean, sorry for the late reply. If you think is ok, I will make the change.

Reply:
This is exactly what I mean. If you accept I will make the changes.

Reply:
Yes, better error-testing of `global_unicast_address()` makes sense. If you re-PR this to `master` with the fixed syntax, I'd approve the PR